### PR TITLE
Add tests to SceneToSaveModelV2 transformer

### PR DESCRIPTION
**Pull request by ivanortegaalba**
**Why?**
To ensure we don't miss any property, and we detect any change that might affect the JSON stored in the DB, I tried to add the most complete possible schema, that reflects what we are able to export nowadays from a DashboardScene

**Next steps**
- Once the transformer in charge of converting a JSON into a Scene is in place, we can run the most complete JSON against it, to see if the Schema -> DashboardScene -> Schema is consistent and it doesn't miss any field in between.
- For any new prop or change in the transformer, this snapshot will detect it

**How to review this PR?**

- [ ] Please make sure the JSON schema contains the needed fields to be stored. If something doesn't make sense, we can modify the transform to only store the really needed properties.
- [ ] Make sure we have added all possible properties from a dashboard to the DashboardScene created in the test to ensure this test covers all properties.

### CODE REVIEW

File: public/app/features/dashboard-scene/serialization/__snapshots__/transformSceneToSaveModelSchemaV2.test.ts.snap [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+        "refresh": "onDashboardLoad",
+        "regex": "regex1",
+        "skipUrlSync": false,
+      },
+    },
+    {
+      "kind": "ConstantVariable",
+      "spec": {
+        "current": {
+          "text": undefined,
```
Comment (harisrozajac) [inline public/app/features/dashboard-scene/serialization/__snapshots__/transformSceneToSaveModelSchemaV2.test.ts.snap:180]: I'm wondering why this one is undefined when it's required as a part of https://github.com/grafana/grafana/blob/6d3a4853b3ef2ff964ef44febc6d96dd2e0a7fbf/packages/grafana-schema/src/schema/dashboard/v2alpha0/dashboard.gen.ts#L902

**Review by harisrozajac — APPROVED** [review]
This looks good and if we are missing something we can add to it. These are things I noticed on the top level of the schema:
- `refresh` not in v2
-  `schemaVersion` not in v2

it could be that some of these have been renamed in v2, like `graphTooltip` becoming `cursorSync`, or we are not planning to include them in v2

**Review by dprokop — APPROVED** [review]
LGTM. There's a valid comment from @harisrozajac.

File: public/app/features/dashboard-scene/serialization/__snapshots__/transformSceneToSaveModelSchemaV2.test.ts.snap (hunk 1 above)
Comment (dprokop) [reply public/app/features/dashboard-scene/serialization/__snapshots__/transformSceneToSaveModelSchemaV2.test.ts.snap:180]: The thing is we are not setting the `text` property for the ConstantVariable (and probably the interval neither). So in the transformer function i think we should either use `value` for the text of the persisted model, or see if we didn't do any special magic for these two variable types in the old architecture. But I don't think so as these don't allow text representation configuration.

Comment (ivanortegaalba) [reply public/app/features/dashboard-scene/serialization/__snapshots__/transformSceneToSaveModelSchemaV2.test.ts.snap:180]: Solved. In the old arch we were using value

**Pull request merged**